// instagram-notes/dist/index.mjs
// src/index.ts
import axios from "axios";
import qs from "query-string";

// src/config.json
var baseURL = "https://i.instagram.com/api/v1/notes/";
var loginURL = "https://i.instagram.com/api/v1/accounts/login/";
var fetchHeaderURL = "https://i.instagram.com/api/v1/si/fetch_headers/?challenge_type=signup";
var defaultHeaders = {
  "accept-language": "en-US",
  authority: "www.instagram.com",
  "content-type": "application/x-www-form-urlencoded",
  origin: "https://www.instagram.com",
  "sec-fetch-site": "same-origin",
  "sec-fetch-mode": "cors",
  "sec-fetch-dest": "empty",
  "user-agent": "Instagram 10.8.0 Android (18/4.3; 320dpi; 720x1280; Xiaomi; HM 1SW; armani; qcom; en_US)",
  "x-ig-app-id": 936619743392459,
  "x-ig-www-claim": "hmac.AR3W0DThY2Mu5Fag4sW5u3RhaR3qhFD_5wvYbOJOD9qaPjIf",
  "x-instagram-ajax": 1,
  "x-requested-with": "XMLHttpRequest"
};
var loginHeaders = {
  "User-Agent": "Instagram 10.8.0 Android (18/4.3; 320dpi; 720x1280; Xiaomi; HM 1SW; armani; qcom; en_US)",
  "Content-Type": "application/x-www-form-urlencoded",
  "Accept-Language": "en-US"
};

// src/index.ts
var Note = class {
  sessionId;
  constructor(sessionId) {
    this.sessionId = `sessionid=${sessionId}`;
  }
  static formatCookie(setCookie) {
    return setCookie?.map((x) => x.match(/(.*?=.*?);/)?.[1])?.join("; ");
  }
  static async getSessionId(username, password) {
    const fetchHeaders = (await axios.get(fetchHeaderURL)).headers;
    const cookie = this.formatCookie(fetchHeaders["set-cookie"]) || "";
    const payload = qs.stringify({
      username,
      password,
      device_id: crypto.randomUUID(),
      login_attempt_count: 0
    });
    const { headers } = await axios.post(loginURL, payload, {
      headers: { ...loginHeaders, cookie }
    });
    const sessionIdCookie = headers["set-cookie"]?.find(
      (e) => e.startsWith("sessionid=")
    );
    return sessionIdCookie?.match(/sessionid=([A-Za-z0-9%_]+)/)?.[1] || "";
  }
  async privateRequest(method, url, data, headers) {
    return await axios.request({
      url,
      method,
      baseURL,
      data,
      headers: {
        ...defaultHeaders,
        ...headers,
        Cookie: this.sessionId
      }
    });
  }
  async getNotes() {
    const { data } = await this.privateRequest("GET", "get_notes/");
    return data;
  }
  async createNote(text, audience = 0) {
    const { data } = await this.privateRequest("POST", "create_note/", {
      audience,
      text,
      uuid: crypto.randomUUID()
    });
    return data;
  }
  async deleteNote(id) {
    const { data } = await this.privateRequest("POST", "delete_note/", {
      id,
      uuid: crypto.randomUUID()
    });
    return data;
  }
  async lastSeenUpdateNote() {
    const { data } = await this.privateRequest(
      "POST",
      "update_notes_last_seen_timestamp/",
      { uuid: crypto.randomUUID() }
    );
    return data;
  }
};
var src_default = Note;
export {
  src_default as default
};
